lightning/src/ln/onchaintx.rs

```diff
@@ -594,17 +594,15 @@ impl<ChanSigner: ChannelKeys> OnchainTxHandler<ChanSigner> {
 								chan_utils::get_revokeable_redeemscript(&chan_keys.revocation_key, *on_counterparty_tx_csv, &chan_keys.broadcaster_delayed_payment_key)
 							};
 
-							if let Ok(sig) = self.key_storage.sign_justice_transaction(&bumped_tx, i, *amount, &per_commitment_key, htlc, &self.secp_ctx) {
-								bumped_tx.input[i].witness.push(sig.serialize_der().to_vec());
-								bumped_tx.input[i].witness[0].push(SigHashType::All as u8);
-								if htlc.is_some() {
-									bumped_tx.input[i].witness.push(chan_keys.revocation_key.clone().serialize().to_vec());
-								} else {
-									bumped_tx.input[i].witness.push(vec!(1));
-								}
-								bumped_tx.input[i].witness.push(witness_script.clone().into_bytes());
-							} else { return None; }
-							//TODO: panic ?
+							let sig= self.key_storage.sign_justice_transaction(&bumped_tx, i, *amount, &per_commitment_key, htlc, &self.secp_ctx).expect("sign justice tx");
```

Comment:
I assume you meant docs on `generate_claim_tx` - this is done.

Reply:
Oops, sorry I missed this, no, I meant on `sign_holder_commitment_and_htlcs` - `generate_claim_tx` isn't pub (so the docs here aren't visible), and the panic impacts all the way up to users (as it can be called in a number of contexts related to `ChannelMonitor`). Alternatively, we could note it in `ChannelMonitor` in all the `pub` functions that call through to here, but the call graph isn't exactly trivial.

Reply:
Added comment on `sign_holder_commitment_and_htlcs`